### ISSUE TITLE
first frame for videos in next slides?

### ISSUE THREAD

**Issue by ousia** [Cimbali/pympress#262]
@Cimbali,

this [PDF document](https://github.com/Cimbali/pympress/files/7475984/uncompressed-a.pdf) contains a video on the second slide.

When the slide isn’t being the current one, there is no indication that the slide contains a video:

![next-slide-fs8](https://user-images.githubusercontent.com/203767/209791791-447be987-1200-4fc8-8225-d5cac13095ef.png)

Only when in current, video is played:

![video-playing-fs8](https://user-images.githubusercontent.com/203767/209791787-193291f1-9da2-4da1-a1b9-7870eff3c9c9.png)

Would it be possible that when slides contain videos the first frame could be displayed as next slides?

Many thanks for your help.

**Comment by ousia** [comment]
I have just noticed that slides may embed audio in slides with text and/or images.

This uncompressed [PDF document with audio on second slide](https://github.com/Cimbali/pympress/files/10313504/uncompressed-audio.pdf) gives the following:

![una-now-fs8](https://user-images.githubusercontent.com/203767/209797901-481b38c0-0e1d-4859-8234-4125c562e856.png)

Maybe a way to do this would be to add a play sign:

![uncomp-audio-fs8](https://user-images.githubusercontent.com/203767/209797963-f2258ec8-aead-4972-8d6d-047411d54ac0.png)

Just in case it might help (it could be a workaround for video too), here you have the SVG for the “media inside” sign:
 
![media-inside](https://user-images.githubusercontent.com/203767/209798019-5bd875ce-196e-4a2a-883c-fccff5b066ad.svg)

Many thanks for your help.

**Comment by ousia** [comment]
To test the button, you may use this [PDF](https://github.com/Cimbali/pympress/files/10314242/slides-with-colors.pdf).

**Comment by ousia** [comment]
The new button is here:

![media-play](https://user-images.githubusercontent.com/203767/209813634-b9995b5f-f62c-4094-af7a-420317b66056.svg)

I think it is a visible option for most of the color combinations (I have tested 😅).

Many thanks for your help.

**Comment by Cimbali** [comment]
Thanks for your input @ousia !

> Would it be possible that when slides contain videos the first frame could be displayed as next slides?

> I have just noticed that slides may embed audio in slides with text and/or images.

So on this, I think it depends on whether the video is specifying:
1. a placeholder (e.g. an image or something),
2. the first frame of the video (typically done with a video started and immediately paused)
3. nothing

I think it’s probably not great to add anything in situation 1., and in situation 2. we can reasonably only do this in the current slide, so adding a frame looking like a play button like you propose would be reasonable on the next-slides. In situation 3. I’m not entirely sure, the question is what the use case is: do people just not bother to put a placeholder but want one? Or do people intentionally want a transparent video frame that can then overlay other content? Anything we do in this latest case has to be presenter-side only and not content-side. Maybe if the overlay is very transparent (like 20%?) we can add it anyway?

**Comment by Cimbali** [comment]
You’re right I haven’t looked at the spec to figure out if there’s something defined there. However it doesn’t look like annotations’ appearance characteristics dictionaries (the `MK` field in `Screen` annotations) are accessible currently through Poppler. I’m already having [a hard time getting additional action info merged](https://gitlab.freedesktop.org/poppler/poppler/-/merge_requests/1295) even by writing it all out myself… I think for now we have to figure out a sensible default that does not rely on extracting more info from the PDF, unfortunately.

Basically, I’m worried about people intentionally setting an intentionally-transparent playable annotation on top of other content. Is it OK, or too annoying, to set an overlay on top like you propose (on the presenter side only)? This also should take into account that we currently don’t implement “poster” or “first frame” display.

Also zero-sized rects are kind of annoying -- if you don’t have autoplay, then there’s no way of activating that annotation, as there is no real area on which to click. Even with the button you propose, I thought of imagined it being on the area defined by the annotation rect. Otherwise it gets really messy as soon as you have e.g. 2 sounds on the same page.

Starting from #273 we no longer get a black rectangle on annotations playing audio files, does that mean (if you’re generating these PDFs yourself) that you could now have these annotations have a non-zero area?

**Comment by Cimbali** [comment]
> I intend the overlaid media logo to be a warning for the presenter

I understand, but I still would prefer to only add something on the annotation area.
